code/modules/events/vent_clog.dm

```diff
@@ -0,0 +1,294 @@
+	playsound(spawn_location, 'sound/effects/splat.ogg', 30, TRUE)
+
+///Signal catcher for plunger_act()
+/datum/round_event/vent_clog/proc/plunger_unclog(datum/source, obj/item/plunger/attacking_plunger, mob/user, reinforced)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(attempt_unclog), user)
+	return COMPONENT_NO_AFTERATTACK
+
+///Handles the actual unclogging action and ends the event on completion.
+/datum/round_event/vent_clog/proc/attempt_unclog(mob/user)
+	if(vent.welded)
+		to_chat(user, span_notice("You cannot pump [vent] if it's welded shut!"))
+		return
+
+	to_chat(user, span_notice("You begin pumping [vent] with your plunger."))
+	if(do_after(user, 6 SECONDS, target = vent))
+		to_chat(user, span_notice("You finish pumping [vent]."))
+		clear_signals()
+		kill()
+
+///Handles the initial steps of clogging a vent, either at event start or when the vent moves.
+/datum/round_event/vent_clog/proc/clog_vent()
+	RegisterSignal(vent, COMSIG_PARENT_QDELETING, PROC_REF(vent_move))
+	RegisterSignal(vent, COMSIG_PLUNGER_ACT, PROC_REF(plunger_unclog))
+
+	for(var/turf/nearby_turf in view(2, get_turf(vent)))
+		if(isopenturf(nearby_turf) && prob(85))
+			new /obj/effect/decal/cleanable/dirt(nearby_turf)
+
+	produce_mob()
+
+///Clears the signals related to the event, before we wrap things up.
+/datum/round_event/vent_clog/proc/clear_signals()
+	UnregisterSignal(vent, list(COMSIG_PARENT_QDELETING, COMSIG_PLUNGER_ACT))
+
+/datum/round_event_control/vent_clog/major
+	name = "Ventilation Clog: Major"
+	typepath = /datum/round_event/vent_clog/major
+	weight = 12
+	max_occurrences = 5
+	earliest_start = 10 MINUTES
+	description = "Dangerous mobs climb out of a vent."
+	min_wizard_trigger_potency = 0
+	max_wizard_trigger_potency = 4
+
+/datum/round_event/vent_clog/major/setup()
+	. = ..()
+	maximum_spawns = rand(MOB_SPAWN_MINIMUM, 5)
+	spawn_delay = rand(15,20)
+	filth_spawn_types = list(
+		/obj/effect/decal/cleanable/blood,
+		/obj/effect/decal/cleanable/insectguts,
+		/obj/effect/decal/cleanable/fuel_pool,
+		/obj/effect/decal/cleanable/oil,
+	)
+
+/datum/round_event/vent_clog/major/get_mob()
+	var/static/list/mob_list = list(
+		/mob/living/basic/mouse/rat,
+		/mob/living/simple_animal/hostile/bee,
+		/mob/living/basic/giant_spider,
+		/mob/living/basic/cockroach/hauberoach,
+	)
+	return pick(mob_list)
+
+/datum/round_event/vent_clog/major/announce()
+	priority_announce("Major biological obstruction detected in the ventilation network. Blockage is believed to be in the [get_area_name(vent)] area.", "Infestation Alert")
+
+/datum/round_event_control/vent_clog/critical
+	name = "Ventilation Clog: Critical"
+	typepath = /datum/round_event/vent_clog/critical
+	weight = 8
+	min_players = 15
+	max_occurrences = 3
+	earliest_start = 25 MINUTES
+	description = "Really dangerous mobs climb out of a vent."
+	min_wizard_trigger_potency = 3
+	max_wizard_trigger_potency = 6
+
+/datum/round_event/vent_clog/critical/setup()
+	. = ..()
+	spawn_delay = rand(15,25)
```

Comment:
```suggestion
	spawn_delay = rand(1.5, 2.5) SECONDS
```

Reply:
This value is a measure of event ticks, not deciseconds.